components/tag-list/tag-list.js

```diff
@@ -34,8 +34,14 @@ class TagList extends LocalizeCoreElement(InteractiveMixin(ArrowKeysMixin(LitEle
 		return {
 			/**
 			 * Enables the option to clear all inner tag list items. The `d2l-tag-list-item-clear` event will be dispatched for each list item when the user selects to Clear All. The consumer must handle the actual item deletion.
+			 * @type {boolean}
 			 */
 			clearable: { type: Boolean },
+			/**
+			 * ADVANCED: When an item is `clearable`, optionally add a timeout before the focus happens on clear. This is useful if the consumer has some operations that will reload the list items prior to wanting focus to occur.
+			 * @type {number}
+			 */
+			clearFocusTimeout: { type: Number, attribute: 'clear-focus-timeout' },
```

Comment:
Does the delay feel bad?  If design doesn't like it, we could rework filter to send the change event immediately for `requestFilterValueClear`.  I remember noticing it in the labs component, but I didn't care because it was labs 😛

Reply:
Hmm is there a functional reason the delay would be needed or is it just to make changes happen a bit slower?

Reply:
The delay is there for when someone is clicking around in the filter quickly, we batch the changes up for perf benefits.  But we already have a short-circuit in place that sends the event immediately if the user presses the clear or clear all button, and we could hook into that.

Keeping the delay would help for perf if the user were to delete a bunch of tags at once, so it's a tradeoff.

Reply:
Ah gotcha. I'm okay with keeping this if it helps with perf. It does seem a bit potentially fragile though, so if we do keep it, I should probably comment that the usage in `filter-tags` corresponds to that specific delay in filter so if one changes the other should as well.